### ISSUE TITLE
Extension still working?

### ISSUE THREAD

**Issue by samplereality** [leonkt/zotero-memento#7]
Has the Internet Archive blocked this extension somehow? I've noticed in the past few weeks that I receive an "Archive URL Not Found" error.

Here's the debug output:

```
[JavaScript Error: "Component returned failure code: 0x80470002 (NS_BASE_STREAM_CLOSED) [nsIConverterOutputStream.writeString]" {file: "chrome://zotero/content/xpcom/server.js" line: 601}]

version => 5.0.89, platform => Win32, oscpu => Windows NT 10.0; WOW64, locale => en-US, appName => Zotero, appVersion => 5.0.89, extensions => ZotFile (5.0.16, extension), Zotero Memento (1.0.0, extension), Zotero LibreOffice Integration (5.0.23.SA.5.0.89, extension), Zotero Word for Windows Integration (5.0.16.SA.5.0.89, extension), Shockwave Flash (32.0.0.403, plugin)

(4)(+0000000): SELECT IA.itemID FROM itemAttachments IA NATURAL JOIN items I LEFT JOIN itemData ID ON (IA.itemID=ID.itemID AND fieldID=1) LEFT JOIN itemDataValues IDV ON (ID.valueID=IDV.valueID) WHERE parentItemID=? AND linkMode NOT IN (3) AND IA.itemID NOT IN (SELECT itemID FROM deletedItems) ORDER BY contentType='application/pdf' DESC, value=? DESC, dateAdded ASC [19956, 'https://www.theatlantic.com/politics/archive/2016/05/white-people-dont-use-white-emoji/481695/']

(4)(+0000012): SELECT IA.itemID FROM itemAttachments IA NATURAL JOIN items I LEFT JOIN itemData ID ON (IA.itemID=ID.itemID AND fieldID=1) LEFT JOIN itemDataValues IDV ON (ID.valueID=IDV.valueID) WHERE parentItemID=? AND linkMode NOT IN (3) AND IA.itemID NOT IN (SELECT itemID FROM deletedItems) ORDER BY contentType='application/pdf' DESC, value=? DESC, dateAdded ASC [19956, 'https://www.theatlantic.com/politics/archive/2016/05/white-people-dont-use-white-emoji/481695/']

(4)(+0000006): SELECT IA.itemID FROM itemAttachments IA NATURAL JOIN items I LEFT JOIN itemData ID ON (IA.itemID=ID.itemID AND fieldID=1) LEFT JOIN itemDataValues IDV ON (ID.valueID=IDV.valueID) WHERE parentItemID=? AND linkMode NOT IN (3) AND IA.itemID NOT IN (SELECT itemID FROM deletedItems) ORDER BY contentType='application/pdf' DESC, value=? DESC, dateAdded ASC [19956, 'https://www.theatlantic.com/politics/archive/2016/05/white-people-dont-use-white-emoji/481695/']

(4)(+0000005): SELECT IA.itemID FROM itemAttachments IA NATURAL JOIN items I LEFT JOIN itemData ID ON (IA.itemID=ID.itemID AND fieldID=1) LEFT JOIN itemDataValues IDV ON (ID.valueID=IDV.valueID) WHERE parentItemID=? AND linkMode NOT IN (3) AND IA.itemID NOT IN (SELECT itemID FROM deletedItems) ORDER BY contentType='application/pdf' DESC, value=? DESC, dateAdded ASC [19956, 'https://www.theatlantic.com/politics/archive/2016/05/white-people-dont-use-white-emoji/481695/']

(3)(+0000005): MIME type text/html can be handled natively

(4)(+0000003): SELECT IA.itemID FROM itemAttachments IA NATURAL JOIN items I LEFT JOIN itemData ID ON (IA.itemID=ID.itemID AND fieldID=1) LEFT JOIN itemDataValues IDV ON (ID.valueID=IDV.valueID) WHERE parentItemID=? AND linkMode NOT IN (3) AND IA.itemID NOT IN (SELECT itemID FROM deletedItems) ORDER BY contentType='application/pdf' DESC, value=? DESC, dateAdded ASC [19956, 'https://www.theatlantic.com/politics/archive/2016/05/white-people-dont-use-white-emoji/481695/']

(4)(+0000532): SELECT IA.itemID FROM itemAttachments IA NATURAL JOIN items I LEFT JOIN itemData ID ON (IA.itemID=ID.itemID AND fieldID=1) LEFT JOIN itemDataValues IDV ON (ID.valueID=IDV.valueID) WHERE parentItemID=? AND linkMode NOT IN (3) AND IA.itemID NOT IN (SELECT itemID FROM deletedItems) ORDER BY contentType='application/pdf' DESC, value=? DESC, dateAdded ASC [19956, 'https://www.theatlantic.com/politics/archive/2016/05/white-people-dont-use-white-emoji/481695/']

(5)(+0003889): CookieSandbox: Being paranoid about channel for cors-anywhere.herokuapp.com

(5)(+0000000): CookieSandbox: Cleared cookies to be sent to cors-anywhere.herokuapp.com

(5)(+0007228): CookieSandbox: Being paranoid about channel for cors-anywhere.herokuapp.com

(5)(+0000002): CookieSandbox: No Set-Cookie header received for cors-anywhere.herokuapp.com
```

**Comment by Mela** [comment]
I'm having the same problem. First I thought, I maybe blew a limit by adding 200-300 objects to my zotero database in once night. But if I'm not the only one, maybe the plugiu is broken.

**Comment by beaugunderson** [comment]
Also seeing this error, though not 100% of the time (I can try an import multiple times and sometimes it works eventually).

**Comment by thtbln** [comment]
"Archive URL Not Found" error seems to be persistent. However, when I search a couple of minutes later for the URL in the wayback machine, in most cases I find a saved snapshot which seems to be created at the time when the message about successful archiving appears.

**Comment by Tetizera-zz** [comment]
Same problem here. I also want to know if the extension is still supported and working.

**Comment by wilkohardenberg** [comment]
I'm also wondering if it still works: I import items in Zotero and then click on Archive this Resource, but the plugin doesn't seem to do anything at all. No archive pages are created and, consequently, nothing is recorded in the extra field. The error console tells me that no chrome package is registered for overlay.dtd (apologies, I seem to be unable to copy and paste from the console)

**Comment by leonkt** [comment]
This was something made as a summer project. I'm a currently a student with little time to resolve this; perhaps I'll fix this at some time in the future.

**Comment by salim-b** [comment]
@leonkt Thanks for your straightforward answer. Would a bug bounty help motivate you (i.e. assign some of your spare time) to fix the extension? E.g. via [Bountysource ](https://www.bountysource.com/) (see the [FAQ](https://github.com/bountysource/core/wiki/Frequently-Asked-Questions) for an intro)?

**Comment by leonkt** [comment]
That won't be necessary haha. I'll set aside some time during the next break to see if I can resolve this.

**Comment by element4l** [comment]
Nothing seems to happen when I click "Archive this resource > Internet Archive." No window, no popup, no note added, no info added in extra, no error. It's as if nothing happened.

**Comment by leonkt** [comment]
After a while, I got around to fixing it! :) New version under Releases, use the instructions previously listed to install.

**Comment by salim-b** [comment]
> After a while, I got around to fixing it! :) New version under Releases, use the instructions previously listed to install.

Thank you very much!

I've just installed the latest release and it does add archive links again to newly added items. But the add-on still completely "freezes" Zotero's UI until the archiving is done – which sometimes takes up to several minutes (see also https://github.com/leonkt/zotero-memento/issues/9)... 😐 I understand that this is probably due to the slow archive.org API. Thus, it would be awesome if you somehow could make the extension's archiving process _non-blocking_!

**Comment by leonkt** [comment]
I released a version, v1.1.1, where this freezing bug should be fixed!

**Comment by shawnmjones** [comment]
In case anyone comes here looking for help, we recently rewrote this extension as "Zotero Robust Links": https://github.com/lanl/Zotero-Robust-Links-Extension
